test/linalg/matmul.jl

```diff
@@ -164,6 +165,21 @@ let A = [1+2im 3+4im; 5+6im 7+8im], B = [2+7im 4+1im; 3+8im 6+5im]
     end
 end
 
+#vecdot for generic reals as well
+let A = [1 3; 5 7], B = [2 4; 3 6]
+    @test vecdot(A,B) == dot(vec(A),vec(B)) == vecdot_(A,B) == vecdot(float(A),float(B))
+    @test vecdot(Int[], Int[]) == 0 == vecdot_(Int[], Int[])
+    @test_throws MethodError vecdot(Any[], Any[])
```

Comment:
I think you're right. I'll remove this for now and we can come back to it later.